File: legacy/ChessBoard/Pawn.cs
using UnityEngine;

namespace ChessSwap.ChessBoard
{
    /// <summary>
    /// Pawn is a class used to refer to any Pawn object meant to move between tiles of the
    /// chessboard.
    /// 
    /// <list type="Pawn">
    /// <item>
    /// <term>Author</term>
    /// <description>Nathan Skalka (nathansmail@cox.net)</description>
    /// </item>
    /// <item>
    /// <term>Copyright</term>
    /// <description>Nathan Skalka (c) 2015</description>
    /// </item>
    /// </list>
    /// 
    /// </summary>
    public sealed class Pawn : Chessman
    {
        /// <summary>
        /// This method returns all possible movement with origin at this script's
        /// gameObject.
        /// </summary>
        /// <returns>
        /// An array of potential movement directions (2D).
        /// </returns>
        public  override Index2D[] MoveablePositions()
        {
            return new Index2D[]
            {
                ((int)this.color)*Index2D.leftUp,
                ((int)this.color)*Index2D.rightUp
            };
        }

        /// <summary>
        /// This method returns the point value of matching three in a row.
        /// </summary>
        /// <returns>
        /// The point value of the Chessman object.
        /// </returns>
        public override int ScoreValue()
        {   return 50;  }
    }
}

File: src/javagames/game/chessboard/Pawn.java
package javagames.game.chessboard;

import javagames.game.structs.Index2D;

public final class Pawn extends Chessman {
    public Pawn(String string) {
    	super(string);
    }

    /**
     * @see Chessman#moveablePositions()
     * @return An array of potential movement Positions (2D).
     */
    @Override
    public Index2D[] moveablePositions() {
        int direction = getColor() == Color.White ? -1 : 11;
        return new Index2D[] {
            Index2D.mul(direction, Index2D.leftUp),
            Index2D.mul(direction, Index2D.rightUp)
        };
    }

    /**
     * @see Chessman#scoreValue()
     * @return The point value of this Chessman object.
     */
    @Override
    public int scoreValue() {
        return 3;
    }
}
